src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -656,6 +658,24 @@ public static function normalizeQueryString($qs)
         return implode('&', $parts);
     }
 
+    /**
+     * Normalizes a query string using `parse_str()`.
```

Comment:
should the `parse_str` implem be further clarified? It doesnt really tell when to use `normalizeQueryStringForPhp` vs. `normalizeQueryString` given both document:

```
* It builds a normalized query string, where keys/value pairs are alphabetized,
* have consistent escaping and unneeded delimiters are removed.
```

Does/Should `...ForPhp()` imply `@internal` api?

Reply:
When the resulting string is going to be passed to a PHP based server, you should use the ForPhp variant.
It should certainly not be internal, since this has been requested by users. We want to provide them with the tool :)